integration/log.go

```diff
@@ -304,70 +303,63 @@ func readbackLogEntries(logID int64, client trillian.TrillianLogClient, params T
 		}
 		return r
 	}(expect)
-
-	currentLeaf := int64(0)
-	leafMap := make(map[int64]*trillian.LogLeaf)
 	glog.Infof("Expecting %d unique leaves", len(expect))
 
-	for currentLeaf < params.LeafCount {
-		// We have to allow for the last batch potentially being a short one
-		numLeaves := params.LeafCount - currentLeaf
-
-		if numLeaves > params.ReadBatchSize {
-			numLeaves = params.ReadBatchSize
+	var leaves []*trillian.LogLeaf
+	for index, end := params.StartLeaf, params.StartLeaf+params.LeafCount; index < end; {
+		// We have to allow for the last batch potentially being a short one.
```

Comment:
IMO this (pre-existing) comment doesn't add a lot, but perhaps I've read/written a lot of batched retrieval stuff and this is useful to someone that hasn't encountered this pattern before?